src/results.cpp

```diff
@@ -215,7 +215,13 @@ util::Optional<RowExpr> Results::first()
         case Mode::Query:
         case Mode::TableView:
             update_tableview();
-            return m_table_view.size() == 0 ? util::none : util::make_optional(m_table_view.front());
+            if (m_table_view.size() == 0) {
+                return util::none;
+            } else if (m_update_policy == UpdatePolicy::Never && !m_table_view.is_row_attached(0)) {
```

Comment:
Minor nit, but in object store we typically don't use braces on single-line if statements. It's also not really necessary to have an `else` when the `if`'s body ended with a `return`.